modular_zubbers/code/modules/job_interns/job_interns.dm

```diff
@@ -0,0 +1,269 @@
+/// Enables the choice of players disabling their Intern ID tag
+/datum/preference/toggle/be_intern
+	category = PREFERENCE_CATEGORY_GAME_PREFERENCES
+	savefile_key = "be_intern"
+	savefile_identifier = PREFERENCE_PLAYER
+	default_value = TRUE
+
+/datum/job
+	/// Whether the ID of the job can be tagged as an intern at all
+	var/can_be_intern = TRUE
+	/// Whether the job uses its own EXP to define the internship status
+	var/internship_use_self_exp_type = FALSE
+	/// What department EXP to use to determine internship
+	var/internship_dep_exp_type = null
+	/// What master job to use for EXP checks if `/datum/config_entry/flag/use_intern_master_job_unlock_threshold` config is enabled
+	var/intern_master_job = null
+
+/datum/job/proc/get_intern_time_threshold()
+	if(!internship_dep_exp_type)
+		return 0
+	var/config_type
+	switch(internship_dep_exp_type)
+		if(EXP_TYPE_COMMAND)
+			config_type = /datum/config_entry/number/intern_threshold_command
+		if(EXP_TYPE_SECURITY)
+			config_type = /datum/config_entry/number/intern_threshold_security
+		if(EXP_TYPE_SUPPLY)
+			config_type = /datum/config_entry/number/intern_threshold_cargo
+		if(EXP_TYPE_ENGINEERING)
+			config_type = /datum/config_entry/number/intern_threshold_engineering
+		if(EXP_TYPE_SERVICE)
+			config_type = /datum/config_entry/number/intern_threshold_service
+		if(EXP_TYPE_MEDICAL)
+			config_type = /datum/config_entry/number/intern_threshold_medical
+		if(EXP_TYPE_SCIENCE)
+			config_type = /datum/config_entry/number/intern_threshold_science
+	if(!config_type)
+		return 0
+	return global.config.Get(config_type) * 60
+
+/// Returns whether a player should be tagged as an intern in this job
+/datum/job/proc/player_joins_as_intern(client/player_client)
+	if(!can_be_intern)
+		return FALSE
+	if(!CONFIG_GET(flag/allow_intern_job_tags))
+		return FALSE
+	if(!SSdbcore.Connect())
+		return FALSE
+	var/player_allows_intern = player_client.prefs.read_preference(/datum/preference/toggle/be_intern)
+	if(!player_allows_intern)
```

Comment:
Much better to clarify with a comment than to spend time on caching and holding a variable

```suggestion
	if(!player_client.prefs.read_preference(/datum/preference/toggle/be_intern) // If the pref is off, we stop here
```